src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -166,26 +166,64 @@ export class AccessibleBufferWidget extends DisposableStore {
 		this._editorWidget.focus();
 	}
 
-	async createQuickPick(): Promise<IQuickPick<IAccessibleBufferQuickPickItem> | undefined> {
-		this._cursorPosition = withNullAsUndefined(this._editorWidget.getPosition());
+	navigateToCommand(type: 'next' | 'previous'): void {
+		const currentLine = this._editorWidget.getPosition()?.lineNumber || this._getDefaultCursorPosition()?.lineNumber;
+		const commands = this._getCommandsWithEditorLine();
+		if (!commands?.length || !currentLine) {
+			return;
+		}
+		const filteredCommands = type === 'previous' ? commands.filter(c => c.lineNumber < currentLine).sort((a, b) => b.lineNumber - a.lineNumber) : commands.filter(c => c.lineNumber > currentLine).sort((a, b) => a.lineNumber - b.lineNumber);
```

Comment:
nit: You could make this faster by using reduce, it's a pretty long line currently

Reply:
I always find reduce functions confusing to read and it won't be that many imo to make it worth that